From f20b9ca45d4b1658be604f83424b334a5b896984 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sat, 8 Feb 2014 21:18:12 +0000
Subject: [PATCH 6/6] CDRipper: fix the writing of the metadata to the tag

The encoders only update the tag in their dtor but the ripper only deletes the
encoder when starting encoding a new track by this time the track has been
copied to the 'Music' storage group so we now delete the encoder just before
starting the copy to the storage group.
---
 mythplugins/mythmusic/mythmusic/cdrip.cpp |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/mythplugins/mythmusic/mythmusic/cdrip.cpp b/mythplugins/mythmusic/mythmusic/cdrip.cpp
index 21d41c1..dc30a5b 100644
--- a/mythplugins/mythmusic/mythmusic/cdrip.cpp
+++ b/mythplugins/mythmusic/mythmusic/cdrip.cpp
@@ -379,6 +379,9 @@ void CDRipperThread::run(void)
                 titleTrack->setFileSize((quint64)QFileInfo(outfile).size());
                 titleTrack->dumpToDatabase();
 
+                // this will delete the encoder which will write the metadata in it's dtor
+                encoder.reset();
+
                 // copy track to the BE
                 destFile = gCoreContext->GenMythURL(url.host(), 0, destFile, "Music");
 
-- 
1.7.10

